src/simulation/hdf_simulation_store.jl

```diff
@@ -119,6 +116,7 @@ end
 function Base.close(store::HdfSimulationStore)
     flush(store)
     HDF5.close(store.file)
+    empty!(store.cache)
```

Comment:
I added this change because I think we should free the user's memory. Let me know if disagree. We have a different mechanism to cache results in SimulationResults.

Reply:
I am ok with this. The cache shouldn't be populated after closing.